File: example-python/main/town.py
class Town:
    """
    Represents a town with name and number of residents.

    :param str name: The name of the town.
    :param int residents: The number of residents in the town.
    """

    def __init__(self, name, residents):
        self.name = name
        self.residents = residents

    @property
    def name(self):
        """The name of the town."""
        return self.__name

    @name.setter
    def name(self, value):
        if value and not value.isspace():
            self.__name = value
        else:
            raise ValueError("Name cannot be null, empty or contain only white spaces.")

    @property
    def residents(self):
        """The number of residents in the town."""
        return self.__residents

    @residents.setter
    def residents(self, value):
        self.__residents = value

    def __str__(self):
        """Returns a human readable string of the object."""
        return f"The town {self.name} has {self.residents} residents."

    def __repr__(self):
        """Returns a string representation of the object."""
        return f"{type(self).__name__}({self.name!r}, {self.residents!r})"

File: example-java/src/main/java/example/Town.java
package example;

/**
 * Represents a town with name and number of residents.
 */
public class Town {

  /** The name of the town. */
  private String name;

  /** The number of residents in the town. */
  private int residents;

  public Town(String name, int residents) {
    this.setName(name);
    this.setResidents(residents);
  }

  public String getName() {
    return this.name;
  }

  public int getResidents() {
    return this.residents;
  }

  public void setName(String name) {
    if (name == null || name.isBlank())
      throw new IllegalArgumentException("Name cannot be null, empty or contain only white spaces.");
    this.name = name;
  }

  public void setResidents(int residents) {
    this.residents = residents;
  }

  /** Returns the summarized information of the town. */
  public String getInfo() {
    return "The town " + this.name + " has " + this.residents + " residents.";
  }

  @Override
  /** Returns a string representation of the object. */
  public String toString() {
    return "Town('" + name + "', " + residents + ")";
  }

}
